ios/Exponent/Versioned/Core/UniversalModules/EXScopedFileSystemModule.m

```diff
@@ -0,0 +1,63 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+#import "EXScopedFileSystemModule.h"
+#import <EXFileSystem/EXFileSystem.h>
+#import <EXConstantsInterface/EXConstantsInterface.h>
+
+@interface EXScopedFileSystemModule ()
+
+@property (nonatomic, weak) NSString *appOwnership;
+
+@end
+
+@implementation EXScopedFileSystemModule
+
+- (instancetype)initWithExperienceId:(NSString *)experienceId constantsModule:(id<EXConstantsInterface>)constantsModule
+{
+  self.appOwnership = constantsModule.appOwnership;
+  self = [super initWithExperienceId:experienceId];
+  return self;
+}
+
+- (NSString *)documentDirectoryForExperienceId:(NSString *)experienceId
+{
+  return [EXScopedFileSystemModule documentDirectoryForExperienceId:experienceId isDetached:![self isItExpoClient]];
+}
+
+- (NSString *)cachesDirectoryForExperienceId:(NSString *)experienceId
+{
+  return [EXScopedFileSystemModule cachesDirectoryForExperienceId:experienceId isDetached:![self isItExpoClient]];
+}
+
+- (bool)isItExpoClient
+{
+  return [_appOwnership isEqualToString:@"expo"];
+}
+
++ (NSString *)escapedResourceName:(NSString *)name
+{
+  NSString *charactersToEscape = @"!*'();:@&=+$,/?%#[]";
+  NSCharacterSet *allowedCharacters = [[NSCharacterSet characterSetWithCharactersInString:charactersToEscape] invertedSet];
+  return [name stringByAddingPercentEncodingWithAllowedCharacters:allowedCharacters];
+}
+
++ (NSString *)documentDirectoryForExperienceId:(NSString *)experienceId isDetached:(BOOL)isDetached {
```

Comment:
This method doesn't need to have this argument, it can just always use `[EXEnvironment sharedEnvironment].isDetached`